jdaviz/core/marks.py

```diff
@@ -334,12 +340,6 @@ def __init__(self, viewer, value=0, **kwargs):
     def marks(self):
         return [self, self.label]
 
-    def _on_global_display_unit_changed(self, msg):
-        # Updating the value is handled by the plugin itself, need to update unit string.
-        if msg.axis in ["spectral", "x"]:
-            self.xunit = msg.unit
-        self._update_label()
```

Comment:
the `_update_label` may still be necessary to change the units in the label, but the rest probably can now be replaced with `super()._on_global_display_unit_changed(msg)`.  Sorry for missing that!

Reply:
I'll double check this now

Reply:
I'm seeing that the values for LinearScale and the slice indicator are not (0, 1). I can alter it to just test for LinearScale but I will need to check if there are an adverse affects from this.

If I do alter this and include this change, the label value updates but the unit does not on the slice indicator (which should be feasible to update but is proving to be a bit tricky).